From 79e3ef9b8989f64840096b8826aedb7468a65e00 Mon Sep 17 00:00:00 2001
From: Xinyun Liu <xinyun.liu@intel.com>
Date: Tue, 29 May 2018 22:25:38 +0800
Subject: [PATCH 1041/1240] drm/i915/gvt: Adjust array size avoid out of
 boundary

dom0_regs is one array with 4 elements. But it is possible to use the
index 4 based on plane_owner, which is out of boundary.
Use the same array_size for plane_owner and dom0_regs.

Signed-off-by: Xinyun Liu <xinyun.liu@intel.com>
Reviewed-by: Min He <min.he@intel.com>
Acked-by: Zhao Yakui <yakui.zhao@intel.com>
---
 drivers/gpu/drm/i915/gvt/gvt.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/gvt/gvt.h b/drivers/gpu/drm/i915/gvt/gvt.h
index c4dadcc..c5b220a 100644
--- a/drivers/gpu/drm/i915/gvt/gvt.h
+++ b/drivers/gpu/drm/i915/gvt/gvt.h
@@ -288,8 +288,8 @@ struct intel_gvt_pipe_info {
 	int owner;
 	struct intel_gvt *gvt;
 	struct work_struct vblank_work;
-	struct intel_dom0_plane_regs dom0_regs[I915_MAX_PLANES - 1];
-	int plane_owner[I915_MAX_PLANES - 1];
+	struct intel_dom0_plane_regs dom0_regs[I915_MAX_PLANES];
+	int plane_owner[I915_MAX_PLANES];
 	int scaler_owner[SKL_NUM_SCALERS];
 };
 
-- 
2.7.4

